addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/serializer/executors/XMLSerializer.java

```diff
@@ -0,0 +1,225 @@
+        return readonly != null ? Boolean.parseBoolean(readonly.getValue()) : null;
+    }
+
+    protected Boolean getIsAutoloopFromXMLNode(Element elem) {
+        Attribute autoloop = elem.attribute("autoloop");
+        return autoloop != null ? Boolean.parseBoolean(autoloop.getValue()) : null;
+    }
+
+    protected String getNameFromXMLNode(Element elem) {
+        Attribute name = elem.attribute("name");
+        return name != null ? name.getValue() : null;
+    }
+
+    protected InputType getTypeFromXMLNode(Element elem) {
+        InputType type = null;
+        Attribute typeAtt = elem.attribute("type");
+        if (typeAtt != null) {
+            type = InputType.getByValue(typeAtt.getValue());
+        }
+
+        return type == null ? StringValue : type;
+    }
+
+    @Override
+    public String doSerialization(List<TemplateInput> params) {
+        Element root = DocumentFactory.getInstance().createElement(fieldsTag);
+
+        for (TemplateInput param : params) {
+            Element field = root.addElement(fieldTag);
+            try {
+                doSerialization(field, param);
+            } catch (TemplateInputBadFormat e) {
+                log.error("Can't Serialize the following param: {}", param);
+                root.remove(field);
+            }
+        }
+        return root.asXML();
+    }
+
+    protected void doSerialization(Element field, TemplateInput param) {
+
+        field.addAttribute("name", param.getName());
+
+        InputType type = param.getType();
+
+        if (type == null) {
+            if (param.getStringValue() == null) {
+                log.warn("Null param: {}", param::getName);
+                throw new TemplateInputBadFormat();
+            } else
+                type = StringValue;
+        }
+
+        field.addAttribute("type", type.getValue());
+        if (param.isReadOnly()) {
+            field.addAttribute("readonly", "true");
+        }
+        if (param.isAutoLoop()) {
+            field.addAttribute("autoloop", "true");
+        }
+        if (param.getDesciption() != null) {
+            field.setText(param.getDesciption());
+        }
+
+        switch (type) {
+        case StringValue:
+            field.addAttribute("value", param.getStringValue());
+            break;
+        case DateValue:
+            SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT);
+            field.addAttribute("value", dateFormat.format(param.getDateValue()));
+            break;
+        case BooleanValue:
+            field.addAttribute("value", param.getBooleanValue().toString());
+            break;
+        case MapValue:
+        case ListValue:
+            Map<String, TemplateInput> map = param.getMapValue();
+            for (String childParamName : map.keySet()) {
+                TemplateInput childParam = map.get(childParamName);
+                if (MapValue.equals(type) && childParamName.equals(childParam.getName())) {
```

Comment:
Shouldn't that be negated, `!childParamName.equals(...)` ?
Not sure I get the point of a warn if things still work anyway...